src/core/CorePool.ts

```diff
@@ -157,11 +157,36 @@ export class CorePool {
     amount0Requested: JSBI,
     amount1Requested: JSBI
   ): { amount0: JSBI; amount1: JSBI } {
-    // TODO
     this.checkTicks(tickLower, tickUpper);
+
+    let position: Position = this.positionManager.getPositionAndInitIfAbsent(
+      PositionManager.getKey(recipient, tickLower, tickUpper)
+    );
+
+    let amount0 =
+      amount0Requested > position.tokensOwed0
+        ? position.tokensOwed0
+        : amount0Requested;
+    let amount1 =
+      amount1Requested > position.tokensOwed1
+        ? position.tokensOwed1
+        : amount1Requested;
+
+    if (amount0 > ZERO) {
+      position.updateBurn(
+        JSBI.subtract(position.tokensOwed0, amount0),
+        position.tokensOwed1
+      );
+    }
+    if (amount1 > ZERO) {
+      position.updateBurn(
+        position.tokensOwed1,
+        JSBI.subtract(position.tokensOwed1, amount1)
+      );
+    }
```

Comment:
if (amount0 > ZERO) || (amount1 > ZERO) {
      position.updateBurn(
        JSBI.subtract(position.tokensOwed0, amount0),
        JSBI.subtract(position.tokensOwed1, amount1)
      );
    }

Reply:
Make sense here, am I correct that `amount0` and `amount1` only could be value 0 or value positive?

Reply:
That's a good question, let's assert that at line 161.

Reply:
Ok!